src/util/package-json-reader.ts

```diff
@@ -4,35 +4,28 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import {Result} from '../error.js';
+import {AsyncCache} from './async-cache.js';
+import {PackageJson} from './package-json.js';
 import * as pathlib from 'path';
 import * as fs from 'fs/promises';
 import {parseTree} from './ast.js';
 
-import type {PlaceholderConfig} from '../analyzer.js';
-import type {JsonAstNode} from './ast.js';
-import {Failure} from '../event.js';
-import {Result} from '../error.js';
-
 export const astKey = Symbol('ast');
 
 export interface JsonFile {
```

Comment:
Should `JsonFile` be in `ast.ts`? It's not used by `package-json-reader.ts` anymore.